cfpb/consumerfinance.gov
# Add Ask CFPB Wagtail report

We currently use a custom menu item in Wagtail, "Export Ask data" (pointing to `/admin/export-ask`), to generate a CSV of content and metadata for every Ask page. Being able to export a CSV of Ask data is very useful for some stakeholders, but the code that powers the export is an old [custom script](https://github.com/cfpb/consumerfinance.gov/blob/main/cfgov/ask_cfpb/scripts/export_ask_data.py) that we've needed to maintain over the years. In the time since that script was created, Wagtail has introduced the ability to make custom reports that have all the functionality of our old script but are built into Wagtail, making maintenance easier. We've converted all of our other old custom exports to Wagtail reports, and now we're moving the Ask export to a standard Wagtail report.

---

## Additions

- Views, templates, and Wagtail hooks for an Ask CFPB report
- Page title and meta description to the items in the report
- Unit tests for the new report

## Changes

- Rearranged everything in `test_reports_view.py` to be in some kind of order

## How to test this PR

1. In the Wagtail admin, go to Reports > Ask CFPB.
2. Verify Ask data shows up in that report. If you're using a recent database dump to test, you should have 47 pages of data in the report.
3. Export the report data as a CSV. Again, if you're using recent database dump, you should have around 2,310 rows in that CSV.
4. Export the Ask data the old way, using the "Export Ask data" menu item.
5. Compare the two CSVs. They should have the same number of rows and the same data (other than the two extra columns that were added to the new Wagtail report).

## Notes and todos

- Note that Wagtail clips the horizontal overflow on admin pages, so you probably won't be able to see all the columns in the report unless you make your browser window pretty big. That's just a Wagtail thing that's best fixed upstream (and also people really only want the CSV export anyway).
- I'm leaving the old Ask export in place to (a) keep this PR small and (b) let users test out the new report and confirm for themselves that it's the same as the old custom export. I'll remove the old export script in a subsequent PR.

## Checklist

  - PR titles are used to generate the change log in [releases](../../releases); good ones make that easier to scan.
  - Consider prefixing, e.g., "Mega Menu: fix layout bug", or "Docs: Update Docker installation instructions".
  - [This repo’s docs](https://cfpb.github.io/consumerfinance.gov/) (edit the files in the `/docs` folder) – for basic, close-to-the-code docs on working with this repo
  - CFGOV/platform wiki on GHE – for internal CFPB developer guidance
  - CFPB/hubcap wiki on GHE – for internal CFPB design and content guidance
